test/controllers/api/v2/repository_sets_controller_test.rb

```diff
@@ -103,6 +103,36 @@ def test_available_repositories
       assert_response :success
     end
 
+    def test_available_repo_sort
+      task = assert_sync_task ::Actions::Katello::RepositorySet::ScanCdn
+
+      repo_set = lambda { |version: "", arch: "", path: ""|
+        { :substitutions => { :releasever => version, :basearch => arch }, :path => path }.with_indifferent_access
+      }
+
+      expected_sort = [repo_set.call(version: nil, arch: "x86_64", path: ""),
```

Comment:
since your lambda function provides a default empty path you don't need to set it on your expected sort list

Reply:
Can you tell me what made you choose to use a lambda here?

Reply:
I'm not sure what you are saying here. :)

Maybe that I can just drop the expected_sort altogether? What would hold my final sorted list?

Reply:
I mean you can change your expected values to ex: `repo_set.call(version: nil, arch: "x86_64")` and omit the path since your lambda function is giving `""` by default. I think that'd be preferred here since your test case isn't concerned with the value in `path` at all

Reply:
I see, when you said path, I wasn't thinking of the actual value, "path", but of something else. :)